libnmstate/nm/connection.py

```diff
@@ -51,69 +56,85 @@ def import_by_id(self, con_id=None):
         if con_id:
             self.con_id = con_id
         if self.con_id:
-            self.profile = self._client.get_connection_by_id(self.con_id)
+            self.profile = self._ctx.client.get_connection_by_id(self.con_id)
 
     def update(self, con_profile):
-        user_data = con_profile.profile.get_interface_name()
         flags = NM.SettingsUpdate2Flags.BLOCK_AUTOCONNECT
         flags |= NM.SettingsUpdate2Flags.TO_DISK
+        action = f"Update profile: {self.profile.get_id()}"
+        user_data = action
         args = None
 
-        self._mainloop.push_action(
-            self.profile.update2,
+        self._ctx.register_async(action, fast=True)
+        self.profile.update2(
             con_profile.profile.to_dbus(NM.ConnectionSerializationFlags.ALL),
             flags,
             args,
-            self._mainloop.cancellable,
+            self._ctx.cancellable,
             self._update2_callback,
             user_data,
         )
 
-    def add(self, save_to_disk=True):
+    def add(self):
         nm_add_conn2_flags = NM.SettingsAddConnection2Flags
         flags = nm_add_conn2_flags.BLOCK_AUTOCONNECT
-        if save_to_disk:
-            flags |= nm_add_conn2_flags.TO_DISK
-        else:
-            flags |= nm_add_conn2_flags.IN_MEMORY
+        flags |= nm_add_conn2_flags.TO_DISK
```

Comment:
I do not understand why `save_to_disk` is not valid anymore.

Reply:
I think it is not related to this mainloop change.

Reply:
The `save_to_disk` is only used by test code. I removed it to simplify the work in nm integration test.

Will add it back.